src/main/java/com/st/lms/service/AdminService.java

```diff
@@ -0,0 +1,338 @@
+			if(lb.getBranchName().equals(branchName))
+				return true;
+		}
+		return false;
+	}
+	
+	public boolean branchInfoExists(String branchName, String branchAddr) {
+		List<LibraryBranch> libBranches = null;
+		libBranches = libraryBranchDao.findAll();
+		
+		for(LibraryBranch lb : libBranches) {
+			if(lb.getBranchName().equals(branchName) && lb.getBranchAddress().equals(branchAddr))
+				return true;
+		}
+		return false;
+	}
+	
+	public List<LibraryBranch> getAllBranches() {
+		return libraryBranchDao.findAll();
+	}
+	
+	public LibraryBranch getLibraryBranch(int id) {
+		Optional<LibraryBranch> libraryBranch = libraryBranchDao.findById(id);
+		if(libraryBranch.isPresent())
+			return libraryBranch.get();
+		return null;
+	}
+	
+	public LibraryBranch addLibraryBranch(LibraryBranch libraryBranch) {
+		Optional<LibraryBranch> existingLibraryBranch = libraryBranchDao.findById(libraryBranch.getBranchId());
+		if(!existingLibraryBranch.isPresent()) 
+			return libraryBranchDao.save(libraryBranch);
+		return null;
+	}
+	
+	public LibraryBranch updateLibraryBranch(int id, LibraryBranch libraryBranch) {
+		Optional<LibraryBranch> existingLibraryBranch = libraryBranchDao.findById(id);
+		if(!existingLibraryBranch.isPresent())
+			return null;
+		return libraryBranchDao.save(libraryBranch);
+	}
+	
+	public void deleteLibraryBranch(int id) {
+		libraryBranchDao.deleteById(id);
+	}
+	
+	/*#########################################################################*/
+	
+	public boolean borrowerNameExists(String borrowerName) {
+		List<Borrower> borrs = null;
+		borrs = borrowerDao.findAll();
+		
+		for(Borrower borr : borrs) {
+			if(borr.getName().equals(borrowerName))
+				return true;
+		}
+		return false;
+	}
+	
+	public List<Borrower> getAllBorrowers() {
+		return borrowerDao.findAll();
+	}
+	
+	public Borrower getBorrower(int id) {
+		Optional<Borrower> borrower = borrowerDao.findById(id);
+		if(borrower.isPresent())
+			return borrower.get();
+		return null;
+	}
+	
+	public Borrower addBorrower(Borrower borrower) {
+		Optional<Borrower> existingBorrower = borrowerDao.findById(borrower.getCardNo());
+		if(!existingBorrower.isPresent()) 
+			return borrowerDao.save(borrower);
+		return null;
+	}
+	
+	public Borrower updateBorrower(int id, Borrower borrower) {
+		Optional<Borrower> existingBorrower = borrowerDao.findById(id);
+		if(!existingBorrower.isPresent())
+			return null;
+		return borrowerDao.save(borrower);
+	}
+	
+	public void deleteBorrower(int id) {
+		borrowerDao.deleteById(id);
+	}
+	
+	/*#########################################################################*/	
+	
+	public List<BookLoans> getAllBookLoans() {
+		List<BookLoans> bookLoans = null;
+		bookLoans = bookLoansDao.findAll();
+		return bookLoans;
+	}
+	
+	public void changeDueDate(int bookId, int branchId, int cardNo, Date dateOut, Date dueDate) {
+		BookLoans bl = new BookLoans(bookId, branchId, cardNo, dateOut, dueDate);
+		bookLoansDao.saveAndFlush(bl);;

```

Comment:
The only justifiable reason for an empty statement I know of is as an empty loop body; since that's not the case here, remove the extra semicolon.